build.sbt

```diff
@@ -1,14 +1,76 @@
+// TODO: restrict parallelExecution to tests only (the obvious way to do this using Test scope does not seem to work correctly)
+parallelExecution in Global := false
+
+lazy val commonSettings = Seq(
+  organization := "uk.gov.hmrc",
+  majorVersion := 7,
+  makePublicallyAvailableOnBintray := true,
+  resolvers := Seq(
+    Resolver.bintrayRepo("hmrc", "releases"),
+    Resolver.typesafeRepo("releases")
+  )
+)
+
+lazy val library = (project in file("."))
+  .enablePlugins(SbtAutoBuildPlugin, SbtGitVersioning, SbtArtifactory)
+  .settings(
+    commonSettings,
+    publish := {},
+    publishAndDistribute := {},
+    crossScalaVersions := Seq.empty
+  )
+  .aggregate(
+    playSchedulingPlay25,
+    playSchedulingPlay26,
+    playSchedulingPlay27
+  ).disablePlugins(sbt.plugins.JUnitXmlReportPlugin)
+
+lazy val playSchedulingCommon = Project("play-scheduling-common", file("play-scheduling-common"))
+  .enablePlugins(SbtAutoBuildPlugin, SbtArtifactory)
+  .settings(
+    commonSettings,
+    crossScalaVersions := Seq(scala2_11, scala2_12),
+    libraryDependencies ++= AppDependencies.compileCommon ++ AppDependencies.testCommon
+  )
+
+lazy val playSchedulingPlay25 = Project("play-scheduling-play-25", file("play-scheduling-play-25"))
+  .enablePlugins(SbtAutoBuildPlugin, SbtArtifactory)
+  .settings(
+    commonSettings,
+    unmanagedSourceDirectories in Compile += (playSchedulingCommon / Compile / scalaSource).value,
+    unmanagedSourceDirectories in Test += (playSchedulingCommon / Test / scalaSource).value,
+    scalaVersion := scala2_11,
+    libraryDependencies ++= AppDependencies.compileCommon ++ AppDependencies.compilePlay25 ++ AppDependencies.testCommon ++ AppDependencies.testPlay25,
+    dependencyOverrides ++= AppOverrides.overrides
+  )
+
+lazy val playSchedulingPlay26 = Project("play-scheduling-play-26", file("play-scheduling-play-26"))
+  .enablePlugins(SbtAutoBuildPlugin, SbtArtifactory)
+  .settings(
+    commonSettings,
+    unmanagedSourceDirectories in Compile += (playSchedulingCommon / Compile / scalaSource).value,
+    unmanagedSourceDirectories in Test += (playSchedulingCommon / Test / scalaSource).value,
+    crossScalaVersions := Seq(scala2_11, scala2_12),
+    libraryDependencies ++= AppDependencies.compileCommon ++ AppDependencies.compilePlay26 ++ AppDependencies.testCommon ++ AppDependencies.testPlay26,
+    dependencyOverrides ++= AppOverrides.overrides
+  )
+
+lazy val playSchedulingPlay27 = Project("play-scheduling-play-27", file("play-scheduling-play-27"))
+  .enablePlugins(SbtAutoBuildPlugin, SbtArtifactory)
+  .settings(
+    commonSettings,
+    unmanagedSourceDirectories in Compile += (playSchedulingCommon / Compile / scalaSource).value,
+    unmanagedSourceDirectories in Test += (playSchedulingCommon / Test / scalaSource).value,
+    scalaSource in Compile := (playSchedulingPlay26 / Compile / scalaSource).value,
+    scalaVersion := scala2_12,
```

Comment:
I think we need to document why we are not supporting 2.11 here, since play supports it.
If it's only for reactivemongo, and this library does not depend on it, I think we should just build for 2.11.
I'm a bit confused if this library requires reactivemongo, there seems to be a dependency for test, but not for runtime?

Reply:
Hi @colin-lamed - this library is using mongo-lock, which is depends on simple-reactivemongo. So there isn't a version of mongo-lock for Play 2.7 and Scala 2.11, hence why there isn't a version of this for Play 2.7 and Scala 2.11.